src/main/kotlin/br/com/alura/forum/model/TopicDomain.kt

```diff
@@ -0,0 +1,45 @@
+package br.com.alura.forum.model
+
+import br.com.alura.forum.model.TopicStatus.*
+import java.time.Instant
+import javax.persistence.*
+import javax.persistence.EnumType.STRING
+import javax.persistence.GenerationType.IDENTITY
+
+@Entity
+class Topic(
+        val shortDescription: String,
+        @Lob val content: String,
+        @ManyToOne val owner: User,
+        @ManyToOne val course: Course
+) {
+    @Id
+    @GeneratedValue(strategy = IDENTITY)
+    val id: Long? = null
+    val creationInstant: Instant = Instant.now()
+    val lastUpdate: Instant = Instant.now()
+    @Enumerated(STRING) var status = NOT_ANSWERED
+    @OneToMany(mappedBy = "topic") private val _answers = mutableListOf<Answer>()
+    val answers: List<Answer>
+        get() = _answers
+
+    fun registerNewReply(newReply: Answer) =
```

Comment:
To aqui lembrando da implementação do state aqui... realmente é chato, mas o double dispatch corta os ifs... Digo:

```
status.registerNewReplay(this,answer)
```
ia chamar de volta
```
addReply(answer)
```
Esse último método podia até ter uma visibilidade limitada, para ninguém de fora chamar. Continuo achando que a troca faz sentido, esses ifs, para esse caso, tendem a deixar mais fácil. Talvez menos flexível :)

Reply:
o que eu pensei foi que se fosse pra fazer com o TopicState e colocando lógica no TopicStatus, pra mim faria mais sentido o TopicStatus conseguir se alterar..ser um objeto q muda o valor de si mesmo..faz sentido?

Reply:
@asouza dá sua opinião aqui?

e aproveitando..tem mais alguma melhoria que pensa que dê pra fazer?